tools/scripts/release/crate-bump.sh

```diff
@@ -26,14 +24,38 @@ for word in ${crate_array[@]}; do
     specified_crate_version[$key]=$value
 done
 
-for crate in ${updated_crates[@]}; do
-    version=$RELEASE_VERSION
-    name=$(eval "tomlq package.name -f implementations/rust/ockam/$crate/Cargo.toml")
+declare -A bumped_crates
 
-    if [[ ! -z "${specified_crate_version[$crate]}" ]]; then
-        echo "bumping $crate as ${specified_crate_version[$crate]}"
-        version="${specified_crate_version[$crate]}"
-    fi
+bump_crate() {
+    source tools/scripts/release/crates-to-publish.sh
 
-    echo y | cargo release $version --no-push --no-publish --no-tag --no-dev-version --package $name --execute
-done
+    echo "Bumping crates with updated dependency. Note crates whose version has been updated recently will be omitted"
+    echo "$updated_crates"
+
+    for crate in ${updated_crates[@]}; do
+        version=$RELEASE_VERSION
+        name=$(eval "tomlq package.name -f implementations/rust/ockam/$crate/Cargo.toml")
+
+        if [[ ! -z "${specified_crate_version[$crate]}" ]]; then
+            echo "bumping $crate as ${specified_crate_version[$crate]}"
+            version="${specified_crate_version[$crate]}"
+        fi
+
+        if [[ ! -z "${bumped_crates[$crate]}" ]]; then
+            echo "$crate has been bumped recently"
+            continue
+        fi
+
+        bumped_crates[$crate]=true
+
+        echo "Bumping $crate crate"
+        echo y | cargo release $version --no-push --no-publish --no-tag --no-dev-version --package $name --execute
+    done
+}
+
+bump_crate
+
+# Bump crates that cargo release has modified/updated it's dependencies in `cargo.toml`.
+#
+# Get crates whose cargo.toml file has been updated omitting recently updated crates.
+bump_crate
```

Comment:
Line 56 bumps crates which has been updated since last tag and then we crates whose ockam dependencies are updated.